easybuild/framework/easyblock.py

```diff
@@ -4422,6 +4426,70 @@ def print_dry_run_note(loc, silent=True):
     dry_run_msg(msg, silent=silent)
 
 
+def copy_build_dirs_logs_failed_installs(application_log, silent, app, easyconfig):
+    """
+    Copy build directories and log files for failed installation (if desired)
+    """
+    # there may be multiple log files, or the file name may be different due to zipping
+    logs = glob.glob(f"{application_log}*")
+
+    datestamp = time.strftime('%Y%m%d')
+    timestamp = time.strftime('%H%M%S')
+    salt = ''.join(random.choice(ascii_letters) for i in range(5))
+    unique_subdir = f'{datestamp}-{timestamp}-{salt}'
+
+    operation_args = []
+
+    logs_path = get_failed_installs_logs_path(easyconfig)
+    if logs_path:
+        logs_path = os.path.join(logs_path, unique_subdir)
+
+        if is_parent_path(app.builddir, logs_path):
+            print_warning(
+                "Path to copy log files to for failed installs is subdirectory of build directory; not copying",
+                log=_log,
+                silent=silent
+            )
+        else:
+            operation_args.append(
+                (
+                    copy_file,
+                    logs,
+                    logs_path,
+                    f"Log file(s) of failed installation copied to {logs_path}"
+                )
+            )
+
+    build_dirs_path = get_failed_installs_build_dirs_path(easyconfig)
+    if build_dirs_path and os.path.isdir(app.builddir):
+        build_dirs_path = os.path.join(build_dirs_path, unique_subdir)
+
+        if is_parent_path(app.builddir, build_dirs_path):
+            print_warning(
+                "Path to copy build dirs to for failed installs is subdirectory of build directory; not copying",
+                log=_log,
+                silent=silent
+            )
+        else:
+            operation_args.append(
+                (
+                    lambda src, dest: copy_dir(src, dest, dirs_exist_ok=True),
+                    [app.builddir],
+                    build_dirs_path,
+                    f"Build directory of failed installation copied to {build_dirs_path}"
+                )
+            )
+
+    persistence_paths = [target_path for (_, _, target_path, _) in operation_args]
+    persistence_paths = create_non_existing_paths(persistence_paths)
+
+    for idx, (operation, paths, _, msg) in enumerate(operation_args):
+        for path in paths:
+            operation(path, persistence_paths[idx])
+
```

Comment:
Why the empty line? To me it is easier to see that the print and inner loop are on the same level without that, especially given how short the loop is
```suggestion
```

The same argument holds for the empty line though, but vertical space is more important to me